src/Orc.Theming/ViewModels/BaseColorSchemeSwitcherWithIconViewModel.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace Orc.Theming.ViewModels
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using System.Windows;
+    using Catel;
+    using Catel.IoC;
+    using Catel.MVVM;
+    using Catel.MVVM.Views;
+    using Orc.Theming.Views;
+
+    public class BaseColorSchemeSwitcherWithIconViewModel : ViewModelBase
+    {
+        private readonly IBaseColorSchemeService _baseColorSchemeService;
+        private readonly IServiceLocator _serviceLocator;
+
+#pragma warning disable CA1801 // Review unused parameters
+        public BaseColorSchemeSwitcherWithIconViewModel(IServiceLocator serviceLocator, IBaseColorSchemeService baseColorSchemeService)
+#pragma warning restore CA1801 // Review unused parameters
+        {
+            Argument.IsNotNull(() => serviceLocator);
+            Argument.IsNotNull(() => baseColorSchemeService);
+
+            _serviceLocator = serviceLocator;
+            _baseColorSchemeService = baseColorSchemeService;
+
+            BaseColorSchemes = _baseColorSchemeService.GetAvailableBaseColorSchemes();
+            BaseColorSchemeUris = _baseColorSchemeService.GetAvailableBaseColorSchemeUris();
+
+            SelectedBaseColorScheme = _baseColorSchemeService.GetBaseColorScheme() ?? BaseColorSchemes[0];
+            updateSelectedBaseColorUri();
+        }
+
+        public IReadOnlyList<string> BaseColorSchemes { get; }
+
+        public IReadOnlyList<KeyValuePair<string, string>> BaseColorSchemeUris { get; }
+
+        public string SelectedBaseColorScheme { get; set; }
+
+        public KeyValuePair<string, string> SelectedBaseColorSchemeUri { get; set; }
+
+        public Visibility ShowImages { get; set; }
+
+        public Visibility ShowText { get; set; }
+
+        protected override async Task InitializeAsync()
+        {
+            await base.InitializeAsync();
+
+            _baseColorSchemeService.BaseColorSchemeChanged += OnBaseColorSchemeServiceBaseColorSchemeChanged;
+        }
+
+        protected override async Task CloseAsync()
+        {
+            _baseColorSchemeService.BaseColorSchemeChanged -= OnBaseColorSchemeServiceBaseColorSchemeChanged;
+
+            await base.CloseAsync();
+        }
+
+        private void OnBaseColorSchemeServiceBaseColorSchemeChanged(object sender, EventArgs e)
+        {
+            SelectedBaseColorScheme = _baseColorSchemeService.GetBaseColorScheme();
+        }
+
+        private void OnSelectedBaseColorSchemeChanged()
+        {
+            updateSelectedBaseColorUri();
+            _baseColorSchemeService.SetBaseColorScheme(SelectedBaseColorScheme);
+        }
+
+        private void OnSelectedBaseColorSchemeUriChanged()
+        {
+            updateSelectedBaseColor();
+            _baseColorSchemeService.SetBaseColorScheme(SelectedBaseColorScheme);
+        }
+
+        private void updateSelectedBaseColorUri()
+        {
+            for (var i = 0; i < BaseColorSchemes.Count; i++)
```

Comment:
important: always use { and }, it's too easy to get mixed up when not using them.